internal/commands/scan.go

```diff
@@ -1343,7 +1389,20 @@ func getUploadURLFromSource(cmd *cobra.Command, uploadsWrapper wrappers.UploadsW
 			}
 		}
 
-		zipFilePath, dirPathErr = compressFolder(directoryPath, sourceDirFilter, userIncludeFilter, scaResolver)
+		if containerScanTriggered {
+			containerResolverError := runContainerResolver(cmd, directoryPath)
+			if containerResolverError != nil {
+				if unzip {
+					_ = cleanTempUnzipDirectory(directoryPath)
+				}
+				return "", "", containerResolverError
+			}
+		}
+		if isSingleContainerScanTriggered(cmd) {
```

Comment:
suggestion - check how the code can be more readable/separate the big method for 2 ? separate compress and unzip to init/ not init